test/irb/test_history.rb

```diff
@@ -158,6 +158,27 @@ def test_history_concurrent_use
       end
     end
 
+    def test_history_concurrent_use_not_present
```

Comment:
I'm not sure if this case covers the fix. I ran it against the old implementation and it still passed.

Reply:
I couldn't think of a way to get it to pass with the current `assert_history` helper method, so I changed the test to be more direct, loading the history without the history file existing, writing the history file, then saving the history.  I checked that this failed previously and now passes.